hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -521,6 +521,14 @@ public boolean hasRowId() {
 		return propertySelectable;
 	}
 
+	public String[] getTableNames() {
+		String[] tableNames = new String[ getTableSpan() ];
+		for ( int i = 0; i < tableNames.length; i++ ) {
+			tableNames[ i ] = getTableName(i);
```

Comment:
I don't think we usually pad the index but the parameter should be.

Reply:
fixed